test/extended/util/deployment.go

```diff
@@ -0,0 +1,75 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+// Define test waiting time const
+const (
+	defaultMaxWaitingTime = 200 * time.Second
+	defaultPollingTime    = 2 * time.Second
+)
+
+// GetDeploymentPods gets the pods list of the deployment by labelSelector
+func GetDeploymentPods(oc *CLI, deployName, namespace, labelSelector string) (*corev1.PodList, error) {
+	return oc.AdminKubeClient().CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: ParseLabelsOrDie(labelSelector).String()})
+}
+
+// GetDeploymentTemplateAnnotations gets the deployment template annotations
+func GetDeploymentTemplateAnnotations(oc *CLI, deployName, namespace string) map[string]string {
+	deployment, err := oc.AdminKubeClient().AppsV1().Deployments(namespace).Get(context.Background(), deployName, metav1.GetOptions{})
+	e2e.ExpectNoError(err)
+	return deployment.Spec.Template.Annotations
+}
+
+// WaitForDeploymentReady waits for the deployment become ready
+func WaitForDeploymentReady(oc *CLI, deployName, namespace string) error {
+	var (
+		deployment    *v1.Deployment
+		labelSelector string
+		getErr        error
+	)
+	pollErr := wait.PollUntilContextTimeout(context.Background(), defaultPollingTime, defaultMaxWaitingTime, true, func(context.Context) (isReady bool, err error) {
+		deployment, getErr = oc.AdminKubeClient().AppsV1().Deployments(namespace).Get(context.Background(), deployName, metav1.GetOptions{})
+		if getErr != nil {
+			e2e.Logf("Unable to retrieve deployment/%s\n %v", deployName, getErr)
+		}
+		if deployment.Status.AvailableReplicas == *deployment.Spec.Replicas {
+			e2e.Logf("Deployment/%s is ready", deployName)
+			return true, nil
+		}
+		e2e.Logf("Deployment/%s is still unready, available replicas %s/%s", deployment.Status.AvailableReplicas, *deployment.Spec.Replicas)
```

Comment:
This produces following log:
```
INFO: Deployment/%!s(int32=1) is still unready, available replicas %!s(int32=2)/%!s(MISSING)
```

* deployName should be the second parameter
* use `%d` (or `%v`) for `AvailableReplicas` / `Replicas`

Reply:
Sorry, I missed it, updated, thanks!